gocourse03/mouseMetropolis/main.go

```diff
@@ -0,0 +1,182 @@
+/*
+У зоопарку є окрема кімната «малі гризуни», де знаходиться кілька сімейств кількох видів гризунів. У кожного є чіп з
+його ID і видом. Для них у цій кімнаті побудована мережа лабіринтів. У цьому лабіринті є один датчик, через який
+впродовж дня проходять сотні гризунів, а до ночі вони розходяться по своїх куточках. Для працівників зоопарку треба
+зробити програму, яка буде давати звіт, де знаходився кожний гризун на початку дня і де зупинився
+наприкінці, а також зберігати історію його рухів через датчик. У програмі може бути тип/структура для певного виду
+гризуна, набори (слайси) цих гризунів, і слайс для зберігання руху гризунів. Також стан сектору лабіринту: скільки
+і які гризуни знаходиться там у певний момент. Треба симулювати рух гризунів через головний датчик і датчики секторів.
+Виконати операції видалення, пошуку і додавання в слайд.
+*/
+package main
+
+import (
+	"fmt"
+	"github.com/brianvoe/gofakeit/v7"
+	"math/rand/v2"
+	"time"
+)
+
+type (
+	Sector     string
+	RodentType string
+	FromTo     [2]Sector
+
+	Movement struct {
+		Time time.Time
+		FromTo
+	}
+
+	MiceRoom struct {
+		Rodents []Rodent
+	}
+
+	Rodent struct {
+		ID      int
+		Type    RodentType
+		History []Movement
+	}
+)
+
+const (
+	rodentCount = 15
+
+	rat   RodentType = "rat"
+	mouse RodentType = "mouse"
+
+	sector1 Sector = "sector1"
+	sector2 Sector = "sector2"
+	sector3 Sector = "sector3"
+	sector4 Sector = "sector4"
+	sector5 Sector = "sector5"
+)
+
+func newRodent(id int) *Rodent {
+	typeRodent := mouse
+
+	if rand.N(2) == 1 {
+		typeRodent = rat
+	}
+	return &Rodent{ID: id + 1, Type: typeRodent, History: nil}
+}
+
+func (r *Rodent) addHistory(movement Movement) {
+	r.History = append(r.History, movement)
+}
+
+func (m *MiceRoom) delHistory(t time.Time) {
+	for _, rodent := range m.Rodents {
+		for i, m := range rodent.History {
+			if m.Time == t {
+				fmt.Println("time found and successfully deleted: ", m)
+				rodent.History = append(rodent.History[:i], rodent.History[i+1:]...)
+			}
+		}
+	}
+}
+
+func (m *MiceRoom) findHistory(ID int) {
```

Comment:
Назви змінних в параметрах завжди з маленької літери починаються:
```suggestion
func (m *MiceRoom) findHistory(id int) {
```

Це можна взяти собі на замітку.